feat: add horoscope functionality to tarot readings

olisaagbafor opened the pull request:
## Changes
- Added zodiac sign determination based on birth month/day
- Created new `draws_card_with_horoscope` function
- Extended NFT properties to include horoscope data
- Added new events for horoscope tracking

## Testing
- Existing tests remain unchanged
- New functionality uses same randomness framework

Closes #6

olisaagbafor committed:
feat: add horoscope functionality to tarot readings

- Add zodiac sign calculation based on birthdate
- Integrate horoscope with card readings
- Update NFT properties to include zodiac and horoscope

huisq commented:
great code and fast PR thanks! small issue. with the horoscope reading, there wont be question anymore, just "answer" which can be rephrased as "fortune". (User get their respective horoscope fortune of the day based on their horoscope and tarot card drawn) 
fix: 
1. mint horoscope reading without question, rephrase answer as "fortune"
2. a different NFT collection should be used. New collection name: ART3MIS_HOROSCOPE
3. minted count cannot be added to the existing state.minted as that is for reading. add a view function for calling collection::count for this new collection. 
4. for this collection, token_name shall be: horoscope + date, i.e.: TAURUS_24JAN2025

Much appreciated! @olisaagbafor

olisaagbafor commented:
@huisq sorry 😔 for the late response, I have been sick quite a while now but I will attend to this as soon as possible

olisaagbafor committed:
feat: implement horoscope collection and fortune minting

- Create separate ART3MIS_HOROSCOPE collection
- Add horoscope minting with fortune instead of readings
- Implement proper date formatting for token names (ZODIAC_DDMMMYYYY)
- Add separate counter and view function for horoscope collection
- Remove question property from horoscope NFTs

olisaagbafor committed:
Merge branch 'main' of github.com:olisaagbafor/Art3misOracle-contracts

olisaagbafor committed:
Merge branch 'main' into add-horoscope-to-tarot

olisaagbafor commented:
@huisq , pls you can review

huisq commented:
Thank you for your contribution! Much appreciated for the high quality code
